src/pipeline/etl/models/geographic_locations.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import annotations
+
+from datetime import datetime, date
+from decimal import Decimal
+from typing import Optional
+
+from sqlmodel import SQLModel, Field
+from sqlalchemy import Index
+
+# ----------------------------------------------------------------------
+# Geographic / Location tables
+# ----------------------------------------------------------------------
+
+
+class GeographicLocation(SQLModel, table=True):
+    """Geographic location (can be anonymised)."""
+    __tablename__ = "geographic_locations"
+
+    location_id: Optional[int] = Field(default=None, primary_key=True)
+    street_address_id: Optional[int] = Field(default=None,
+                                             description="Reference to street_addresses.street_address_id")
+                                             # foreign_key="street_addresses.street_address_id")
+    city_id: Optional[int] = Field(default=None,
+                                    description="Reference to cities.city_id")
+                                    # foreign_key="cities.city_id")
+    zip_id: Optional[int] = Field(default=None,
+                                   description="Reference to zips.zip_id")
+                                   # foreign_key="zips.zip_id")
+    county_id: Optional[int] = Field(default=None,
+                                     description="Reference to counties.county_id")
+                                     # foreign_key="counties.county_id")
+    state_id: Optional[int] = Field(default=None,
+                                    description="Reference to states.state_id")
+                                    # foreign_key="states.state_id")
+    region_id: Optional[int] = Field(default=None,
+                                     description="Reference to regions.region_id")
+                                     # foreign_key="regions.region_id")
+    fips_id: Optional[int] = Field(default=None,
+                                    description="Reference to fips.fips_id")
+                                    # foreign_key="fips.fips_id")
+    latitude: Optional[Decimal] = Field(default=None,
+                                         description="Can be null or generalized")
+    longitude: Optional[Decimal] = Field(default=None,
+                                          description="Can be null or generalized")
```

Comment:
We can use GeoAlchemy2 to create a geometry column based on the latitude and longitude https://pypi.org/project/GeoAlchemy2/.. this is again just a note. :)